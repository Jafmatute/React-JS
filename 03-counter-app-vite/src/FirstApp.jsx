import PropTypes from 'prop-types';

const newMessage = 'Josue';
const getMessage = () => 'React';

const FirstApp = ({title, subTitle, name}) => {

    //if (!title) throw new Error('el titulo no existe')

    return (
        <>
            <h1 data-testid="test-title">{title}</h1>
            <p> {subTitle}</p>
            <code>{name}</code>
        </>
    )
}

//PropTypes
FirstApp.propTypes = {
    subTitle: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
}

//DefaultProps --entran primero que los PropTypes
FirstApp.defaultProps = {
    name: 'Josue Flores',
    subTitle: "Hola soy un subtitulo",
    //title: "No hay titulo"
}

export default FirstApp;