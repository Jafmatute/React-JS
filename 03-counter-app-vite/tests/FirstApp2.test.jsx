import {render, screen} from "@testing-library/react";
import FirstApp from "../src/FirstApp.jsx";

describe('Pruebas en el archivo <FirstApp />', ()=> {

    const title = "Hola, Soy Goku";
    const subTtile = '<p> Soy un subtitulo</p>';

    test('debe de hacer mach con el snapshot', ()=> {

        const {container} = render(<FirstApp title={title} />)
        expect(container).toMatchSnapshot();
    })

    test('debe de mostrar el mensaje "Hola, Soy Goku"', ()=> {

        render(<FirstApp title={title} />)
        //screen.debug();
        expect(screen.getByText(title)).toBeTruthy();

    })

    test('debe de mostrar el titulo en un h1', ()=> {

        render(<FirstApp title={title} />)
        expect(screen.getByRole('heading', {level:1}).innerHTML).toContain(title);

    })

    test('debe de mostrar el subtitulo enviado por los props', ()=> {


        render(<FirstApp title={title} subTitle={subTtile} />)
        expect(screen.getByText(subTtile)).toBeTruthy();

    })


})