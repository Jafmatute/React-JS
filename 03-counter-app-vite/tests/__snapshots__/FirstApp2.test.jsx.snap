// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el archivo <FirstApp /> debe de hacer mach con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, Soy Goku
  </h1>
  <p>
     
    Hola soy un subtitulo
  </p>
  <code>
    Josue Flores
  </code>
</div>
`;
